test/support/test_config.h

```diff
@@ -62,9 +62,16 @@
 #define TEST_DPCPP_BACKEND_PRESENT 0
 #endif
 
+#ifdef __SYCL_UNNAMED_LAMBDA__
```

Comment:
Needs to be reviewed with PSTL team also

Reply:
Looks good to me overall. Will it work in both scenario if we replace it with 
```
#define TEST_UNNAMED_LAMBDAS  defined(__SYCL_UNNAMED_LAMBDA__)
```
to reduce number of lines

Reply:
No, it doesn't work, so I left the current version